### PR TITLE
Implementing sip call hold/un-hold functionalities.

### PR DESCRIPTION
**Pull request by martin7890**

### CODE REVIEW

**Comment by ibauersachs** [comment]
@damencho This seems related to Jicofo. Can you please take a look?

File: src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java [hunk 1]
```diff
@@ -160,6 +164,15 @@ static void attachConfigHeaders(
 
                 Header h = request.getHeader(name);
 
+                if (name.equals(SIPHeaderNames.ROUTE) 
+                    || name.equals("P-Asserted-Identity"))
+                {
+                    if (seqNumber != 1)
+                    {
```
Comment (damencho) [inline src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java]: These changes seems unrelated to the PR, could you remove them and squash to commits.

Comment (martin7890) [inline src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java]: Removed the unrelated code and did a commit.

File: src/net/java/sip/communicator/impl/protocol/jabber/extensions/rayo/RayoIqProvider.java [hunk 2]
```diff
@@ -340,6 +362,90 @@ public void setHeader(String name, String value)
...
+            return holdIq;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected void printAttributes(StringBuilder out)
+        {
+            out.toString();
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/protocol/jabber/extensions/rayo/RayoIqProvider.java]: Please remove

File: src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java [hunk 3]
```diff
@@ -137,6 +138,9 @@ static void attachConfigHeaders(
             headerValues.put(name, prefStr);
         }
 
+        CSeqHeader cSeqHeader = (CSeqHeader) request.getHeader(SIPHeaderNames.CSEQ);
+        Long seqNumber = cSeqHeader.getSeqNumber();
+
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java]: Please remove

File: src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 import gov.nist.javax.sip.header.*;
 import net.java.sip.communicator.util.*;
+import net.java.sip.communicator.service.protocol.ProtocolProviderFactory;
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/protocol/sip/ConfigHeaders.java]: Please remove

File: src/net/java/sip/communicator/impl/protocol/jabber/extensions/rayo/RayoIqProvider.java [hunk 5]
```diff
@@ -340,6 +362,90 @@ public void setHeader(String name, String value)
...
+            return unHoldIq;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected void printAttributes(StringBuilder out)
+        {
+            out.toString();
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/protocol/jabber/extensions/rayo/RayoIqProvider.java]: Please remove

**Comment by ibauersachs** [comment]
Please address the comments from the review. And we'd also need you to to sign the [CLA](https://jitsi.org/icla).
